File: java/url-shortener/src/test/java/com/domain/urlshortener/service/UrlShortenerServiceTest.java
package com.domain.urlshortener.service;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit.jupiter.SpringExtension;

/**
 * @author: rocky.hu
 * @date: 2022/4/6 2:08
 */
@ExtendWith(SpringExtension.class)
@SpringBootTest
public class UrlShortenerServiceTest {

    @Autowired
    private UrlShortenerService urlShortenerService;

    @Test
    public void test_createShortUrl() {
        String longUrl = "http://xx.com";
        String shortUrl = urlShortenerService.createShortUrl(longUrl);
    }

    @Test
    public void test_getLongUrl() {
        test_createShortUrl();
        String shortUrl = "https://m.dev.com/G8";
        String longUrl = urlShortenerService.getLongUrl(shortUrl);
    }

}

File: java/src/test/java/com/example/urlshortener/UrlShortenerServiceTest.kt
package com.example.urlshortener

import com.example.urlshortener.model.ShortUrl

import com.example.urlshortener.repository.UrlRepository
import com.example.urlshortener.service.ShortUrlService
import org.junit.jupiter.api.Assertions.assertEquals

import org.mockito.Mockito
import org.junit.jupiter.api.Test;
import org.mockito.junit.MockitoJUnitRunner
import org.mockito.Mockito.`when`
import org.springframework.boot.test.context.SpringBootTest
import java.util.*

@SpringBootTest
class UrlShortenerServiceTest {
    private val mockRepository = Mockito.mock(UrlRepository::class.java)
    private val service = ShortUrlService(mockRepository)

    @Test
    fun testSave() {
        val result = service.save("http://fakercorp.com/fake-long-path-with-many/")
        assertEquals(result, "C6IHIY")
    }

    @Test
    fun testGetOriginalUrl() {
        val ID = "C6IHIY"
        val URL = "http://fakercorp.com/fake-long-path-with-many/"
        val s = ShortUrl().apply {
            id = ID
            originalUrl = URL
        }
        `when`(mockRepository.findById(ID)).thenReturn(Optional.of(s))
        assertEquals(URL, service.getOriginalUrl(ID))
    }
}